flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecChangelogNormalize.java

```diff
@@ -71,20 +75,36 @@
         producedTransformations = StreamExecChangelogNormalize.CHANGELOG_NORMALIZE_TRANSFORMATION,
         minPlanVersion = FlinkVersion.v1_15,
         minStateVersion = FlinkVersion.v1_15)
+@ExecNodeMetadata(
+        name = "stream-exec-changelog-normalize",
+        version = 2,
```

Comment:
Do we really need to introduce a new version ? Because the topology and state layout do not change,

Reply:
> Do we really need to introduce a new version ? Because the topology and state layout do not change,

I also hesitated for a long time on this point. Introducing `StateMetadata` did not actually change the state layout, but the FLIP-190 design document provides an example of upgrading the exec node version after adding a member variable.
> The JSON node of ExecNode A gets an additional property in Flink 1.16:
> 
>  Before:
>
>  {
>    some-prop: 42
>   }
>  After:
>
> {
>  some-prop: 42,
>  some-flag: false
> }
>
> Every plan change will increase the ExecNode version. Since the ExecNode supports both plan layouts, adding an  additional annotation is enough.
>
> @ExecNodeMetadata(name=A, version=1, minPlanVersion=1.15, minStateVersion=1.15)
> @ExecNodeMetadata(name=A, version=2, minPlanVersion=1.16, minStateVersion=1.15)
> class MyExecNode extends ExecNode {...}
> New plans will use the new ExecNode version 2.

Reply:
@luoyuxia @LadyForest If the testing infrastructure would be already in place, I would agree with increasing the version. But we are not testing different versions of ExecNode yet. So increasing the annotation has not effect.

I just noticed that you bumped the ExecNode version of a couple of ExecNodes for the new state metadata properties in CompiledPlan. Your change actually allows having null  for the stateMetadataList , I'm wondering if we should not increase the version for 1.18. We can stay at the old ExecNode version as 1.18 is able to consume 1.17 plans and state.
The reason why noticed that is due to FLINK-32613 where I added an option to the wrong annotation during rebase and no test failed.

Reply:
Hi @twalthr, during the implementation of FLIP-292, I did notice that the current testing for compiled plans is not comprehensive enough, such as FLINK-31884, FLINK-31917, and FLINK-32219. FLIP-292 added `ExecNodeVersionUpgradeSerdeTest` and `TransformationsTest#testUidFlink1_15` to test the deserialization of old plans using the new version, but it may not be sufficient. After reconsideration, I think we can keep the original exec node version until we have a more comprehensive testing framework. I will open a PR and hope you can help review the code. Thank you.

Reply:
> Are we still able to deserialize 1.15 literals with this change?

I suppose so. I adapted the test `TransformationsTest#testUidFlink1_15` to verify the change. The test deserialized the serialized plan using Flink v1.15, where numbers are serialized as string literals.

> Also the SARG values cannot be null anymore?

Sorry, I'm a bit confused on this point. Did you find any incompatibilities?  AFAIK, the changes made on `RexNodeJsonSerializer` does not aim to change the processing logic for `SARG`. You can check `RexNodeJsonSerdeTest` and `SargJsonPlanITCase` for more details.

Reply:
Hi @twalthr, it appears that modifying an already approved FLIP requires a new vote. Considering that feature cut for 1.18 is only a few days away, do you have any suggestions for a quick fix?